compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistModelDefinitionValidator.java

```diff
@@ -135,6 +135,9 @@ public void perform(SyntaxNodeAnalysisContext ctx) {
         try {
             datastore = getDatastore(ctx);
         } catch (BalException e) {
+            if (e.getMessage().contains("the persist.datastore configuration does not exist")) {
+                return;
+            }
```

Comment:
As discussed, let's return null, if the data store config does not exist.

Reply:
Fixed in [19832c0](https://github.com/ballerina-platform/module-ballerina-persist/pull/278/commits/19832c00117d42d97048b443dd2824c6da3ec37d)